src/GoTrueClient.ts

```diff
@@ -1229,13 +1242,37 @@ export default class GoTrueClient {
     if (options?.scopes) {
       urlParams.push(`scopes=${encodeURIComponent(options.scopes)}`)
     }
+    if (options?.isPKCE) {
+      const codeVerifier = this._generatePKCECodeVerifier()
+      setItemAsync(this.storage, this.storageKey, codeVerifier)
+      urlParams.push(`code_verifier=${encodeURIComponent(codeVerifier)}`)
+    }
     if (options?.queryParams) {
       const query = new URLSearchParams(options.queryParams)
       urlParams.push(query.toString())
     }
+
     return `${this.url}/authorize?${urlParams.join('&')}`
   }
 
+  private _generatePKCECodeVerifier() {
+    let code = ''
+    //See PKCE Char Set: https://www.oauth.com/oauth2-servers/pkce/authorization-request/
+    // for details on charset and length of code verifier
+    const minLength = 43
+    const maxLength = 128
+    const characters = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789-._~"
+    const codeLength = Math.random() * (maxLength - minLength) + minLength
+    let counter = 0;
+    while (counter < codeLength) {
+      code += characters.charAt(Math.floor(Math.random() * characters.length));
```

Comment:
`Math.random()` is not a CSPRNG, so I'm reluctant to use it like this. Can you explore the use of https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/generateKey instead? We can fall back to `Math.random()` if this API is not available.

Reply:
Sure, let's go with that for now.

Another option would be to fix a default number  which the user can override